tests/cnf/core/network/metallb/tests/metallb-nodeSelector.go

```diff
@@ -0,0 +1,300 @@
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("MetalLB NodeSelector", Ordered, Label(tsparams.LabelBGPTestCases), ContinueOnFailure, func() {
+	var (
+		nodeAddrList []string
+		addressPool  []string
+		frrk8sPods   []*pod.Builder
+		frrNodeLabel = "app=frr-k8s"
+		err          error
+	)
+
+	const (
+		ipaddressPoolName1    = "ipaddresspool1"
+		ipaddressPoolName2    = "ipaddresspool2"
+		bgpAdvertisementName1 = "bgpadvertisement1"
+		bgpAdvertisementName2 = "bgpadvertisement2"
+	)
+
+	BeforeAll(func() {
+
+		By("Getting MetalLb load balancer ip addresses")
+		ipv4metalLbIPList, ipv6metalLbIPList, err = metallbenv.GetMetalLbIPByIPStack()
+		Expect(err).ToNot(HaveOccurred(), tsparams.MlbAddressListError)
+
+		By("List CNF worker nodes in cluster")
+		cnfWorkerNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+		By("Selecting worker node for BGP tests")
+		workerLabelMap, workerNodeList = setWorkerNodeListAndLabelForBfdTests(cnfWorkerNodeList, metalLbTestsLabel)
+		ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+			APIClient, metav1.ListOptions{LabelSelector: labels.Set(workerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+		err = metallbenv.IsEnvVarMetalLbIPinNodeExtNetRange(ipv4NodeAddrList, ipv4metalLbIPList, nil)
+		Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+		By("Collecting information before test")
+		frrk8sPods, err = pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+			LabelSelector: frrNodeLabel,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Failed to list speaker pods")
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Failed to detect master nodes")
+
+		By("Setting test iteration parameters")
+		_, _, _, nodeAddrList, addressPool, _, err =
+			metallbenv.DefineIterationParams(
+				ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, netparam.IPV4Family)
+		Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+	})
+
+	AfterAll(func() {
+		if len(cnfWorkerNodeList) > 2 {
+			By("Remove custom metallb test label from nodes")
+			removeNodeLabel(workerNodeList, metalLbTestsLabel)
+		}
+	})
+
+	Context("Single IPAddressPool", func() {
+
+		var (
+			err error
+		)
+
+		BeforeAll(func() {
+			By("Creating a new instance of MetalLB Speakers on workers")
+			err = metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+			Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+			waitForWebHookRunning()
```

Comment:
```suggestion
			waitForFRRK8SWebHookRunning()
```

Reply:
After I moved to general afterall no longer needed as a separate func. Removed